js/main.js

```diff
@@ -5,33 +5,96 @@ var $formSubmit = document.querySelector('form');
     notes: $textArea.value,
-    nextEntryId: data.nextEntryId + 1
+    entryId: data.nextEntryId
   };
   data.nextEntryId += 1;
   data.entries.unshift(formObject);
   $setImg.setAttribute('src', 'images/placeholder-image-square.jpg');
   $formSubmit.reset();
+  $ul.prepend(entryDOM(formObject));
+  $newEntry.className = 'hidden';
+  $entriesPage.className = '';
+  data.view = 'entries';
+}
+
+function entryDOM(entry) {
+  var $createLi = document.createElement('li');
+  var $createDivData = document.createElement('div');
+  var $createDivRow = document.createElement('div');
+  var $createImg = document.createElement('img');
+  var $createDivColHalf = document.createElement('div');
+  var $createH2 = document.createElement('h2');
+  var $createP = document.createElement('p');
+
+  $createLi.appendChild($createDivData);
+  $createDivData.setAttribute('data-view', 'entries');
+  $createDivData.appendChild($createDivRow);
+  $createDivRow.setAttribute('class', 'row');
+  $createDivRow.appendChild($createImg);
+  $createImg.setAttribute('src', entry.photoUrl);
+  $createImg.setAttribute('height', '400px');
+  $createImg.setAttribute('class', 'column-half');
+  $createImg.setAttribute('class', 'column-half');
+  $createDivRow.appendChild($createDivColHalf);
+  $createDivColHalf.setAttribute('class', 'column-half');
+  $createDivColHalf.appendChild($createH2);
+  $createH2.textContent = entry.title;
+  $createDivColHalf.appendChild($createP);
+  $createP.textContent = entry.notes;
+  $ul.appendChild($createLi);
+  return $createLi;
 }
 
-// Put the form 's input values into a new object.
-// Add the nextEntryId to the object.
-// Increment the nextEntryId on the data model.
-// Prepend the new object to the entries in the data model.
-// Reset the image preview 's `src'
-// attribute.
-// Reset the form inputs.
+var $entriesPage = document.querySelector('div[data-view="entries"]');
+var $newButton = document.querySelector('.newButton');
+var $newEntry = document.querySelector('div[data-view="entry-form"]');
+var $entriesLink = document.querySelector('a[href="#entry-form"]');
+
+$newButton.addEventListener('click', function () {
+  switchViews($newEntry.getAttribute('data-view'));
+});
+
+$entriesLink.addEventListener('click', function () {
+  switchViews($entriesPage.getAttribute('data-view'));
+});
+
+var $viewElements = document.querySelectorAll('div[data-view]');
+
+function switchViews(view) {
+  for (var i = 0; i < $viewElements.length; i++) {
+    if ($viewElements[i].getAttribute('data-view') !== view) {
+      $viewElements[i].className = 'hidden';
+    } else {
+      $viewElements[i].className = '';
+      data.view = $viewElements[i].getAttribute('data-view');
+    }
+  }
+}
+
+window.addEventListener('DOMContentLoaded', appendDOM);
+var $noItems = document.querySelector('.center-text');
+if ($ul.childElementCount === 0) {
+  $noItems.className = 'center-text hidden';
+} else {
+  $noItems.className = 'center-text';
+}
```

Comment:
You should probably do this in the `appendDOM` function. You might rename that function to something like `renderSavedEntries`